Mathlib/Probability/Independence/Conditional.lean

```diff
@@ -0,0 +1,750 @@
+
+lemma condIndepSet_iff_condIndep (s t : Set Ω) (μ : Measure Ω) [IsFiniteMeasure μ] :
+    CondIndepSet m' hm' s t μ ↔ CondIndep m' (generateFrom {s}) (generateFrom {t}) hm' μ := by
+  simp only [CondIndepSet, CondIndep, kernel.IndepSet]
+
+theorem condIndepSet_iff_condIndepSets_singleton {s t : Set Ω} (hs_meas : MeasurableSet s)
+    (ht_meas : MeasurableSet t) (μ : Measure Ω) [IsFiniteMeasure μ] :
+    CondIndepSet m' hm' s t μ ↔ CondIndepSets m' hm' {s} {t} μ :=
+  kernel.indepSet_iff_indepSets_singleton hs_meas ht_meas _ _
+
+lemma condIndepSet_iff (s t : Set Ω) (hs : MeasurableSet s) (ht : MeasurableSet t)
+    (μ : Measure Ω) [IsFiniteMeasure μ] :
+    CondIndepSet m' hm' s t μ ↔ (μ⟦s ∩ t | m'⟧) =ᵐ[μ] (μ⟦s | m'⟧) * (μ⟦t | m'⟧) := by
+  rw [condIndepSet_iff_condIndepSets_singleton _ _ hs ht μ, condIndepSets_singleton_iff _ _ hs ht]
+
+lemma iCondIndepFun_iff_iCondIndep {β : ι → Type _}
+    (m : ∀ x : ι, MeasurableSpace (β x)) (f : ∀ x : ι, Ω → β x)
+    (μ : Measure Ω) [IsFiniteMeasure μ] :
+    iCondIndepFun m' hm' m f μ
+      ↔ iCondIndep m' hm' (fun x ↦ MeasurableSpace.comap (f x) (m x)) μ := by
+  simp only [iCondIndepFun, iCondIndep, kernel.iIndepFun]
+
+lemma iCondIndepFun_iff {β : ι → Type _}
+    (m : ∀ x : ι, MeasurableSpace (β x)) (f : ∀ x : ι, Ω → β x) (hf : ∀ i, Measurable (f i))
+    (μ : Measure Ω) [IsFiniteMeasure μ] :
+    iCondIndepFun m' hm' m f μ
+      ↔ ∀ (s : Finset ι) {g : ι → Set Ω} (_H : ∀ i, i ∈ s → MeasurableSet[(m i).comap (f i)] (g i)),
+      μ⟦⋂ i ∈ s, g i | m'⟧ =ᵐ[μ] ∏ i in s, (μ⟦g i | m'⟧) := by
+  simp only [iCondIndepFun_iff_iCondIndep]
+  rw [iCondIndep_iff]
+  exact fun i ↦ (hf i).comap_le
+
+lemma condIndepFun_iff_condIndep {β γ : Type _} [mβ : MeasurableSpace β]
+    [mγ : MeasurableSpace γ] (f : Ω → β) (g : Ω → γ) (μ : Measure Ω) [IsFiniteMeasure μ] :
+    CondIndepFun m' hm' f g μ
+      ↔ CondIndep m' (MeasurableSpace.comap f mβ) (MeasurableSpace.comap g mγ) hm' μ := by
+  simp only [CondIndepFun, CondIndep, kernel.IndepFun]
+
+lemma condIndepFun_iff {β γ : Type _} [mβ : MeasurableSpace β] [mγ : MeasurableSpace γ]
+    (f : Ω → β) (g : Ω → γ) (hf : Measurable f) (hg : Measurable g)
+    (μ : Measure Ω) [IsFiniteMeasure μ] :
+    CondIndepFun m' hm' f g μ ↔ ∀ t1 t2, MeasurableSet[MeasurableSpace.comap f mβ] t1
+      → MeasurableSet[MeasurableSpace.comap g mγ] t2
+        → (μ⟦t1 ∩ t2 | m'⟧) =ᵐ[μ] (μ⟦t1 | m'⟧) * (μ⟦t2 | m'⟧) := by
+  rw [condIndepFun_iff_condIndep, condIndep_iff _ _ _ _ hf.comap_le hg.comap_le]
+
+end DefinitionLemmas
+
+section CondIndepSets
+
+variable {m' : MeasurableSpace Ω}
+  [mΩ : MeasurableSpace Ω] [TopologicalSpace Ω] [BorelSpace Ω] [PolishSpace Ω] [Nonempty Ω]
+  {hm' : m' ≤ mΩ} {μ : Measure Ω} [IsFiniteMeasure μ]
+
+@[symm]
+theorem CondIndepSets.symm {s₁ s₂ : Set (Set Ω)}
+    (h : CondIndepSets m' hm' s₁ s₂ μ) : CondIndepSets m' hm' s₂ s₁ μ :=
+  kernel.IndepSets.symm h
+
+theorem condIndepSets_of_condIndepSets_of_le_left {s₁ s₂ s₃ : Set (Set Ω)}
```

Comment:
`le` sounds weird here, as it's just a subset. Since the same terminology is used in the other independence files, let's keep it like that and maybe fix it some other day.